### ISSUE TITLE
No Application was found for the given package name

### ISSUE THREAD

**Issue by kumarashish-tudip** [fastlane/fastlane#10324]
### New Issue Checklist

- [ ✅] Updated fastlane to the latest version
- [ ✅] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [ ✅] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
I installed the fastlane and updated the FastFile according to the steps provided https://docs.fastlane.tools/getting-started/android/setup/ here and then uploaded my first apk on the google play console for the first time manually due to the limitations described in the issue https://github.com/fastlane/fastlane/issues/5283 here.

But while executing the command below it gives an error No application was found for the given package name.

fastlane supply init

There is an existing github issue https://github.com/fastlane/fastlane/issues/3628 for this but according to the solution provided there I have already uploaded my first apk.
##### Complete output when running fastlane, including the stack trace and command used

[fastlane supply init
sh: /Library/Developer/CommandLineTools//usr/bin/xcodebuild: No such file or directory
[22:43:25]: undefined method `split' for nil:NilClass
[!] Google Api Error: applicationNotFound: No application was found for the given package name.]

### Environment

[<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                |
| --------------------------- | ------------------------------------ |
| OS                          | 10.12.5                              |
| Ruby                        | 2.2.4                                |
| Bundler?                    | false                                |
| Git                         | git version 2.11.0 (Apple Git-81)    |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane  |
| Host                        | Mac OS X 10.12.5 (16F73)             |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib           |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016           |
| Is contained                | true                                 |
| Is homebrew                 | false                                |
| Is installed via Fabric.app | false                                |
| Xcode Path                  | /Library/Developer/CommandLineTools/ |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.55.0"

default_platform :android

platform :android do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
  end

  desc "Runs all the tests"
  lane :test do
    gradle(task: "test")
  end

  desc "Submit a new Beta Build to Crashlytics Beta"
  lane :beta do
    gradle(task: "assembleRelease")
    crashlytics

    # sh "your_script.sh"
    # You can also use other beta testing services here
  end

  desc "Deploy a new version to the Google Play"
  lane :deploy do
    gradle(task: "assembleRelease")
    supply
  end

gradle_path = './android/gradlew'
desc “Release apk to play store” 
  lane :playstore do
  gradle(
    task: 'assemble',
    build_type: 'Release'
  )
end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions

# fastlane reports which actions are used. No personal data is recorded. 
# Learn more at https://github.com/fastlane/fastlane#metrics

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
json_key_file "/Users/mm-03/Desktop/eventur/android/GooglePlayAndroidDeveloper-4611355f4f3f.json" # Path to the json secret file - Follow https://github.com/fastlane/supply#setup to get one
package_name "eventurtest" # e.g. com.krausefx.app

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.56.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| CFPropertyList            | 2.3.5        |
| addressable               | 2.5.1        |
| babosa                    | 1.0.2        |
| bigdecimal                | 1.2.6        |
| bundler                   | 1.14.6       |
| claide                    | 1.0.2        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| commander-fastlane        | 4.4.5        |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| domain_name               | 0.5.20170404 |
| dotenv                    | 2.2.0        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.11.0.1     |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| google-api-client         | 0.13.6       |
| googleauth                | 0.5.1        |
| highline                  | 1.7.8        |
| http-cookie               | 1.0.3        |
| httpclient                | 2.8.3        |
| hurley                    | 0.2          |
| io-console                | 0.4.3        |
| json                      | 1.8.1        |
| jwt                       | 1.5.6        |
| libxml-ruby               | 3.0.0        |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| memoist                   | 0.15.0       |
| mime-types                | 3.1          |
| mime-types-data           | 3.2016.0521  |
| mini_magick               | 4.5.1        |
| minitest                  | 5.4.3        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| multipart-post            | 2.0.0        |
| nanaimo                   | 0.2.3        |
| nokogiri                  | 1.5.6        |
| os                        | 0.9.6        |
| plist                     | 3.2.0        |
| power_assert              | 0.2.2        |
| psych                     | 2.0.8        |
| public_suffix             | 2.0.5        |
| rake                      | 10.4.2       |
| rdoc                      | 4.2.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| rouge                     | 1.11.1       |
| rubygems-update           | 2.5.2        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| signet                    | 0.7.3        |
| slack-notifier            | 1.5.1        |
| sqlite3                   | 1.3.13       |
| terminal-notifier         | 1.7.1        |
| terminal-table            | 1.7.3        |
| test-unit                 | 3.0.8        |
| tty-screen                | 0.5.0        |
| uber                      | 0.0.15       |
| unf                       | 0.1.4        |
| unf_ext                   | 0.0.7.4      |
| unicode-display_width     | 1.1.3        |
| word_wrap                 | 1.0.0        |
| xcodeproj                 | 1.5.1        |
| xcpretty                  | 0.2.6        |
| xcpretty-travis-formatter | 0.0.4        |
</details>

*generated on:* **2017-09-13**
</details>]

**Comment by ohayon** [comment]
Hey @kumarashish-tudip, apologies for the issue! Would you mind sharing the entire output from your command line so we can see all of the context involved? Thanks!

**Comment by kumarashish-tudip** [comment]
Hi @ohayon , I have already shared the entire output. It just says no application found and nothing else.

fastlane supply init
sh: /Library/Developer/CommandLineTools//usr/bin/xcodebuild: No such file or directory
[22:43:25]: undefined method `split' for nil:NilClass
[!] Google Api Error: applicationNotFound: No application was found for the given package name.

**Comment by markwk** [comment]
Same error here. Not quite sure how to debug this.

**Comment by ildella** [comment]
same error here. 
Not sure what is happening. 
I've created a new app on Google Play Console, still in draft, and nowhere I am asked about the package name.

fastlane has created Appfile which contains the packagename I already use everywhere in the app, as wel as the json file I've downloaded after setting fastlane up as per https://docs.fastlane.tools/actions/supply/

Now, where does "supply init" look for an app with that package name? 
If on the play console, there's no way it's going to find any.
locally, there is, then probably something is wrong on my local setup and I do not know whay.

**Comment by sannyo** [comment]
I think I am having the same issue/or limitation.

I have multiple app already published in the store under the same acoount, but when I try to update any other app beside the "default" app. (Assume the first app in that account) I get
[!] Google Api Error: applicationNotFound: No application was found for the given package name.

I am pretty sure that the app is there, on the supply summary I can see the right package name provided using package_name, so not sure what the problem might be.

**Comment by jsauve** [comment]
Have you first manually uploaded an AdHoc-signed APK to the Google Play Console? This has to be done first before any subsequent APKs can be sent via Deliver (or any other means that uses the new Google Play deployment mechanism).
